DietPi-Software | VSCodium

nonnorm opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Reference**: https://github.com/MichaIng/DietPi/issues/4267

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-Software | Add VSCodium install option
- DietPi-Software | Make VSCodium require desktop
- DietPi-Software | Also make VSCodium require git

nonnorm committed:
v7.1
+ DietPi-Software | Add VSCodium install option

nonnorm committed:
v7.1
+ DietPi-Software | Make VSCodium require desktop

nonnorm committed:
v7.1
+ DietPi-Software | Make VSCodium also require git

nonnorm commented:
Well, that was simple.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -5496,6 +5504,23 @@ Package: wireguard wireguard-dkms wireguard-tools\nPin: release n=bullseye\nPin-
+			G_EXEC eval "curl -sSfL 'https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo/raw/master/pub.gpg' | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-vscodium.gpg --yes"
+
+			# Install APT repo list
+			G_EXEC eval "echo 'deb https://paulcarroty.gitlab.io/vscodium-deb-rpm-repo/debs vscodium main' > /etc/apt/sources.list.d/vscodium.list"
```
Let's use `dietpi-vscodium.list`. I know we skip the prefix for most lists somehow, but I aim to use it consequently to make identifying easily which files are coming from us and which were created manually or by other installers/packages.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -15362,6 +15387,17 @@ _EOF_
+			Banner_Uninstalling
+
+			G_AGP codium
+			[[ -f '/etc/apt/sources.list.d/vscodium.list' ]] && G_EXEC_NOEXIT=1 G_EXEC rm /etc/apt/sources.list.d/vscodium.list
```
Rename as above.

MichaIng commented:
Lol a pretty simple naked install, but that is all fine when the package-contained setup is good. Good to have this as an alternative to VSCode after #4083.

nonnorm committed:
v7.1
+ DietPi-Software | Rename vscodium.list to dietpi-vscodium.list

Joulinar commented:
and the good thing, version provided via `apt` package is same as available on GitHub. 👍 

From install test
`Setting up codium (1.55.2-1618361277) ...`

on GitHub same version `1.55.2` is available https://github.com/VSCodium/vscodium/releases

Joulinar commented:
I guess it's an issue of `VSCodium` but not sure. Could be as well me doing something wrong. At least I get an error while connecting to GitHub using the official `GitHub Pull Request` extension.

![image](https://user-images.githubusercontent.com/47155374/115027453-059ebd80-9ec4-11eb-905a-7b64900fe905.png)

MichaIng commented:
Please try to install/enable dbus:
```
apt install dbus
systemctl enable --now dbus
```
**EDIT: Ah no that cannot be the reason as it's installed and enabled automatically with all desktop installs.**

MichaIng committed:
v7.1

+ README | Add VSCodium to sources list

MichaIng committed:
v7.1

+ DietPi-Survey_report | Add VSCodium support

MichaIng committed:
v7.1

+ CHANGELOG | VSCodium: As alternative to MS VSCode without MS branding, telemetry and licensing, the VSCodium code editor, with support for various programming languages, including Java, JavaScript, Go, Node.js, Python and C++, has been added as software install option to DietPi.

MichaIng committed:
Merge branch 'dev' into vscodium

nonnorm commented:
It looks like `gnome-keyring` is required, as after that I get no error.

MichaIng commented:
When I try to sign into GitHub from VSCodium, in that status bar, it shows "Signing in to github.com..." forever, while it should show a login prompt, as far as I understand. I was thinking that it might be due to missing default browser, but making Firefox the default browser doesn't change anything 🤔.

Btw, we should add the application icon to the desktop?
_______
**EDIT: Tried to install `gnome-keyring`, but doesn't change something here. LXDE btw.**

Joulinar commented:
You would need to click on `Signing in to github.com...` to open the field where you can enter GitHub token

MichaIng committed:
v7.1

+ DietPi-Software | VSCodium: Create desktop shortcut on install

MichaIng commented:
Ahhh, I didn't recognised that at the top.
**EDIT: Worked for me without `gnome-keyring` as well.**

MichaIng committed:
v7.1

+ DietPi-Software | LXDE: Apply better default icon theme: nuoveXT2 has more individual icons than Adwaita, but lower resolution and overrides the Firefox application icon with an old low res version.

MichaIng committed:
v7.1

+ DietPi-Software | X server: Add framebuffer driver for x86_64 on Stretch as well, as at least on VirtualBox the DRI was not available OOTB
+ DietPi-Software | LXDE: Fix icon theme

Joulinar commented:
@MichaIng 

> EDIT: Worked for me without gnome-keyring as well.

You don't have the error message at the bottom?

MichaIng commented:
Nope.

Joulinar commented:
Hmm I need to do another test. Which extension you used?

MichaIng commented:
The official GitHub PR and Issues extension. I'll retry to be sure.

MichaIng commented:
Okay, it was still required on a fresh setup. I think that daemon is needed to create a new keyring with credentials, while accessing an existing one does not require it, or so.

I'm not sure whether there is an alternative to that package, other issues indicate it, but I couldn't find one when searching packages with -keyring ending at least.

Shall we add it by default or add a note to the docs instead? The package is small, but it's an additional daemon/process running.

Joulinar commented:
depends on what people will use it for. I guess this is dedicated to GitHub? Or does it might be related to other plugins as well?

nonnorm committed:
v7.1
Empty  commit to test VSCodium

nonnorm committed:
v7.1
+ DietPi-Software | Uninstall cleanly

nonnorm committed:
v7.1
+ DietPi-Software | Fix removal

nonnorm commented:
I think that it would probably be best to add it, as you will probably need it for signing in to Microsoft accounts as well.  However, we could probably configure the keychain automatically with the global software password.

MichaIng commented:
I guess so. Cannot imagine that this extension uses its very own credentials/authentication method but that it's an API of codium that is use by other extensions as well.

MichaIng committed:
v7.1

+ DietPi-Software | VSCodium: Install gnome-keyring as dependency, as at least extensions use it to store credentials. It's a backend daemon for the "Secrets API" and there is not really an alternative available, currently: https://freedesktop.org/wiki/Specifications/secret-storage-spec/

MichaIng commented:
Little research, `gnome-keyring` provides a backend daemon for the "Secrets API", described here: https://freedesktop.org/wiki/Specifications/secret-storage-spec/
Aside of the KDE wallet with GUI, there do not seem to be any relevant alternatives currently. Found https://github.com/grawity/secretsd, but it does not encrypt passwords, is hence more an experiment, I'd say.

Good to know about this in general, as other desktop applications might use that API as well to store or receive passwords, and also good to know that this is a central storage, hence you do not only trust the application like VSCodium here (which stores the master password) to safely store your GitHub token or other credentials but the Gnome keyring and API developers as well. Usually not relevant, but quite an important details for security enthusiasts 😉.

nonnorm commented:
Hm, bash doesn't seem to be expanding the second `home/*` glob when uninstalling, and treating it literally.

MichaIng commented:
Hmm , generally works fine here:
```
# bash -c 'rm -v /{root,home/*}/test'
rm: cannot remove '/root/test': No such file or directory
removed '/home/test/test'
```
I'll try it with VSCodium uninstall as well.

nonnorm commented:
As a comparison:
```
# bash -c 'rm -v /{root,home/*}/test'
rm: cannot remove '/root/test': No such file or directory
rm: cannot remove '/home/*/test': No such file or directory
```
I'll try with a fresh VM.
**Edit**: It seems the glob only expands when the file/directory it's looking for exists, otherwise it prints it out literally.

MichaIng committed:
v7.1

+ DietPi-Software | VSCodium: Do not print an error when on uninstall files do not exist that are about to be removed

MichaIng commented:
> It seems the glob only expands when the file/directory it's looking for exists, otherwise it prints it out literally.

Ah yes of course. Taking the glob literal is basically part of the expansion. `*` matches the literal `*` as well. What we want is to hide "file not found" errors: https://github.com/MichaIng/DietPi/pull/4276/commits/02c8bb833a60889450083855b1cf47727223c733

Joulinar reviewed (approved):
going to approve the PR from inside `vscodium`
